### ISSUE TITLE
Clarify role of Scope and rename Namespace to Registry

### ISSUE THREAD

**Issue by stevvooe** [docker/distribution#1338]
This likely belongs as part of the interface refactoring, but we should probably rename [`Namespace`](https://github.com/docker/distribution/blob/67d3675d555e6c46f371c5690599920cb17871ab/registry.go#L26) back to `Registry`, as we aren't really using in that manner.

We may also want to move `Scope` to the `reference` package and make it useful for ascertaining whether or not a `Reference` belongs to a particular `Scope`.

cc @aaronlehmann @dmcgowan

**Comment by dmcgowan** [comment]
I agree with adding `Scope` to the reference package. This should be a straightforward change. As part of our effort to make the registry be able to handle multiple namespaces, changing `Namespace` to `Registry` I don't think makes sense. The name `ScopedRegistry` is likely more appropriate. We should include this change as part of a larger refactor to support this feature.

**Comment by stevvooe** [comment]
> This should be a straightforward change. As part of our effort to make the registry be able to handle multiple namespaces, changing Namespace to Registry I don't think makes sense.

I just don't think the current object represents what we want a namespace to be. It acts as a registry for every use case. The current layout basically says a namespace returns a registry, which I am not sure if a useful abstraction, as we don't use it that way in the daemon.

If we did refactor the daemon to perform registry access through namespace, I could get behind the current model.

**Comment by jstoja** [comment]
@stevvooe Is this still required for the 2.7 release? It doesn't seem like this has been planned for this release (not discussed for +2years). cc @dmcgowan
